src/pages/learn/[id].tsx

```diff
@@ -74,7 +73,33 @@ function LearnDetail({ highlightData }: { highlightData: HighlightData[] }) {
     return { x: x + x * 0.5, y: y + y * 0.5 };
   };
 
+  //하이라이트 인덱스 구하는 함수입니다.
+  //클릭된 하이라이트의 부모 노드와 해당 하이라이트의 텍스트를 인자로 받습니다.
+  //부모노드를 순회하면서 하이라이트 텍스트와 같은 내용이 나올때까지 글자수를 카운트하면서 증가시킵니다.
+  const [highlighIdx, setHighlightIndex] = useState<number>(0);
+  const getHighlightIndex = (parentNode: ParentNode | null, givenString: string) => {
+    if (parentNode?.childNodes) {
+      let stringLength = 0;
+      for (let i = 0; i < parentNode?.childNodes.length; i++) {
+        if (parentNode?.childNodes[i].textContent === givenString) {
+          setHighlightIndex(stringLength);
+          break;
+        }
+        //as number 빼면 undefined 뜨길래... 붙여줬어용
+        stringLength += parentNode?.childNodes[i]?.textContent?.length as number;
```

Comment:
```suggestion
        stringLength += parentNode?.childNodes[i]?.textContent?.length ?? 0;
```

as는 타입을 강제로 지정하는 것이기 때문에 저는 as 사용을 지양하고 있어용
undefined 때문에 이렇게 한 거라면 `?? 0` 처리를 하는 것을 추천합니다 👍🏻

Reply:
오 몰랐네요 감사합니다!